core/transaction/src/lib.rs

```diff
@@ -328,6 +330,29 @@ pub enum Transaction {
     Receipt(ReceiptTransaction),
 }
 
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+pub enum TransactionStatus {
+    Unknown,
+    Completed,
+    Failed,
+}
+
+impl Default for TransactionStatus {
+    fn default() -> Self {
```

Comment:
i don't think we should have a default for this

Reply:
why not?

Reply:
unknown implies something specifically that we have tried to find the status of the transaction but it is unknown or cannot currently be computed. i would rather be explicit about declaring it unknown when necessary.